sdk/android/skin/src/main/java/com/ooyala/android/skin/util/FullScreenHelper.java

```diff
@@ -0,0 +1,178 @@
+package com.ooyala.android.skin.util;
+
+import android.animation.Animator;
+import android.animation.AnimatorListenerAdapter;
+import android.animation.AnimatorSet;
+import android.animation.ObjectAnimator;
+import android.graphics.Point;
+import android.graphics.Rect;
+import android.view.View;
+import android.view.animation.DecelerateInterpolator;
+import android.widget.FrameLayout;
+
+/**
+ * FullScreenHelper is a class developed to support a full screen mode with RecyclerView
+ */
+
+public class FullScreenHelper {
+	private static final int ANIMATION_DURATION = 300;
+
+	private FrameLayout expandedLayout;
+	private FrameLayout currentParentLayout;
+	private Animator currentAnimator;
+	private FrameLayout currentPlayerLayout;
+
+	public FullScreenHelper(FrameLayout expandedLayout) {
+		this.expandedLayout = expandedLayout;
+	}
+
+	public void collapsePlayerLayout() {
+		if (expandedLayout.getVisibility() != View.VISIBLE || currentAnimator != null) {
+			return;
+		}
+
+		final Point globalOffset = new Point();
+		final Rect startBounds = new Rect();
+		final Rect finalBounds = new Rect();
+
+		setFinalBounds(finalBounds, globalOffset);
+
+		expandedLayout.getGlobalVisibleRect(startBounds);
+		expandedLayout.getGlobalVisibleRect(finalBounds, globalOffset);
+		startBounds.offset(-globalOffset.x, -globalOffset.y);
+		final float startScaleFinal = getStartScale(startBounds, finalBounds);
+
+		AnimatorSet set = new AnimatorSet();
+		set
+				.play(ObjectAnimator.ofFloat(expandedLayout, View.X, startBounds.left))
+				.with(ObjectAnimator.ofFloat(expandedLayout, View.Y, startBounds.top))
+				.with(ObjectAnimator
+						.ofFloat(expandedLayout, View.SCALE_X, startScaleFinal))
+				.with(ObjectAnimator
+						.ofFloat(expandedLayout, View.SCALE_Y, startScaleFinal));
+		set.setDuration(ANIMATION_DURATION);
+		set.setInterpolator(new DecelerateInterpolator());
+		set.addListener(new AnimatorListenerAdapter() {
+			@Override
+			public void onAnimationEnd(Animator animation) {
+				cancelAnimation();
+			}
+
+			@Override
+			public void onAnimationCancel(Animator animation) {
+				cancelAnimation();
+			}
+		});
+		set.start();
+		currentAnimator = set;
+
+	}
+
+	public void expandPlayerLayout(FrameLayout playerLayout) {
+		if (expandedLayout.getVisibility() != View.INVISIBLE) {
+			return;
+		}
+
+		if (currentAnimator != null) {
+			currentAnimator.cancel();
+		}
+
+		currentPlayerLayout = playerLayout;
+		currentParentLayout = (FrameLayout) playerLayout.getParent();
```

Comment:
What if the `playerLayout.getParent()` does not return a `FrameLayout`? Can you add a validation here? If we don't get a FrameLayout this will break the app.